semgrep/tests/e2e/snapshots/test_check/test_nested_pattern_either_rule/results.json

```diff
@@ -12,25 +12,7 @@
         "lines": "    nested_patterns_func('foo', 1)",
         "message": "test",
         "metadata": {},
-        "metavars": {
-          "$X": {
-            "abstract_content": "'foo'",
-            "end": {
-              "col": 31,
-              "line": 2,
-              "offset": 48
-            },
-            "start": {
-              "col": 26,
-              "line": 2,
-              "offset": 43
-            },
-            "unique_id": {
-              "md5sum": "<masked in tests>",
-              "type": "AST"
-            }
-          }
-        },
+        "metavars": {},
```

Comment:
and why semgrep-new does not have those metavars, for once?

Reply:
This test is weird, basically

```
➜  e2e git:(EJ-optimizations) ✗ cat rules/nested-pattern-either.yaml
rules:
- id: test
  message: test
  severity: WARNING
  languages: [javascript]
  pattern-either:
    - pattern-either:
      - pattern: nested_patterns_func('foo', ...)
      - pattern: nested_patterns_func('bar', ...)
    - pattern-either:
      - pattern: nested_patterns_func($X, 1, ...)
      - pattern: nested_patterns_func($X, 2, ...)
➜  e2e git:(EJ-optimizations) ✗ cat targets/basic/nested-patterns.js 
function test() {
    nested_patterns_func('foo', 1)
    nested_patterns_func('bar', 2)
    nested_patterns_func('bar', 3)
    nested_patterns_func('baz', 1)
}
```

So binding or not depends on which pattern matches

Reply:
got it